pkg/rest/server.go

```diff
@@ -239,3 +358,150 @@ func sendError(w http.ResponseWriter, message string, err error) {
 		Message: fmt.Sprintf("%s: %v", message, err),
 	})
 }
+
+// MessagesLastMinute holds the count of produced and consumed messages in the last minute and last second
+type MessagesLastMinute struct {
+	Topic          string `json:"topic"`
+	ProducedCount  int64  `json:"produced_count"`
+	ConsumedCount  int64  `json:"consumed_count"`
+	ProducedPerSec int64  `json:"produced_per_sec"`
+	ConsumedPerSec int64  `json:"consumed_per_sec"`
+}
+
+// OffsetHistoryEntry stores offset and timestamp
+type OffsetHistoryEntry struct {
+	Offset    int64
+	Timestamp time.Time
+}
+
+// In-memory rolling window for offsets (not production safe)
+var producedHistory = make(map[string][]OffsetHistoryEntry)
+var consumedHistory = make(map[string][]OffsetHistoryEntry)
+
+// In-memory store for last offsets and timestamps (for demo purposes, not production safe)
+
+// Handler for messages per minute
+func (s *Server) handleMessagesPerMinute(w http.ResponseWriter, r *http.Request) {
+	admin, err := sarama.NewClusterAdminFromClient(s.kafkaClient)
+
+	if err != nil {
+		sendError(w, "Failed to create admin client", err)
+		return
+	}
+
+	// defer admin.Close()
+
+	topics, err := admin.ListTopics()
+
+	if err != nil {
+		sendError(w, "Failed to list topics", err)
+		return
+	}
+
+	// Calculate produced and consumed message counts in the last minute for each topic
+	counts := []MessagesLastMinute{}
+	now := time.Now()
+	totalAllProduced := int64(0)
+	totalAllConsumed := int64(0)
+	totalAllProducedSec := int64(0)
+	totalAllConsumedSec := int64(0)
+
+	for name := range topics {
+		// Produced: sum latest offsets across all partitions
+		partitions, err := s.kafkaClient.Partitions(name)
+		if err != nil {
+			continue
+		}
+		var totalProduced int64 = 0
+		for _, partition := range partitions {
+			offset, err := s.kafkaClient.GetOffset(name, partition, sarama.OffsetNewest)
+			if err == nil {
+				totalProduced += offset
+			}
+		}
+
+		// Consumed: sum committed offsets for all consumer groups
+		var totalConsumed int64 = 0
+		groups, err := admin.ListConsumerGroups()
+		if err == nil {
+			for group := range groups {
+				offsets, err := admin.ListConsumerGroupOffsets(group, map[string][]int32{name: partitions})
+				if err == nil && offsets.Blocks != nil {
+					for _, partition := range partitions {
+						block := offsets.GetBlock(name, partition)
+						if block != nil && block.Offset > 0 {
+							totalConsumed += block.Offset
+						}
+					}
+				}
+			}
+		}
+
+		// Store offset history for rolling window
+		producedHistory[name] = append(producedHistory[name], OffsetHistoryEntry{Offset: totalProduced, Timestamp: now})
+		consumedHistory[name] = append(consumedHistory[name], OffsetHistoryEntry{Offset: totalConsumed, Timestamp: now})
+
+		// Remove entries older than 1 minute
+		pruneMinute := now.Add(-1 * time.Minute)
+		for len(producedHistory[name]) > 0 && producedHistory[name][0].Timestamp.Before(pruneMinute) {

```

Comment:
there is a handy package that can do the filtering for you since go1.23 https://pkg.go.dev/slices#Collect